velox/connectors/clp/tests/ClpConnectorTest.cpp

```diff
@@ -0,0 +1,311 @@
+  const std::string kClpConnectorId = "test-clp";
+
+  void SetUp() override {
+    OperatorTestBase::SetUp();
+    connector::registerConnectorFactory(
+        std::make_shared<connector::clp::ClpConnectorFactory>());
+    auto clpConnector =
+        connector::getConnectorFactory(
+            connector::clp::ClpConnectorFactory::kClpConnectorName)
+            ->newConnector(
+                kClpConnectorId,
+                std::make_shared<config::ConfigBase>(
+                    std::unordered_map<std::string, std::string>{
+                        {"clp.split-source", "local"}}));
+    connector::registerConnector(clpConnector);
+  }
+
+  void TearDown() override {
+    connector::unregisterConnector(kClpConnectorId);
+    connector::unregisterConnectorFactory(
+        connector::clp::ClpConnectorFactory::kClpConnectorName);
+    OperatorTestBase::TearDown();
+  }
+
+  exec::Split makeClpSplit(const std::string& splitPath) {
+    return exec::Split(
+        std::make_shared<ClpConnectorSplit>(kClpConnectorId, splitPath));
+  }
+
+  RowVectorPtr getResults(
+      const core::PlanNodePtr& planNode,
+      std::vector<exec::Split>&& splits) {
+    return exec::test::AssertQueryBuilder(planNode)
+        .splits(std::move(splits))
+        .copyResults(pool());
+  }
+
+  static std::string getExampleFilePath(const std::string& filePath) {
+    std::string current_path = fs::current_path().string();
+    return current_path + "/examples/" + filePath;
+  }
+};
+
+TEST_F(ClpConnectorTest, test1NoPushdown) {
+  auto plan = PlanBuilder()
+                  .startTableScan()
+                  .outputType(
+                      ROW({"requestId", "userId", "method"},
+                          {VARCHAR(), VARCHAR(), VARCHAR()}))
+                  .tableHandle(std::make_shared<ClpTableHandle>(
+                      kClpConnectorId, "test_1", nullptr))
+                  .assignments({
+                      {"requestId",
+                       std::make_shared<ClpColumnHandle>(
+                           "requestId", "requestId", VARCHAR(), true)},
+                      {"userId",
+                       std::make_shared<ClpColumnHandle>(
+                           "userId", "userId", VARCHAR(), true)},
+                      {"method",
+                       std::make_shared<ClpColumnHandle>(
+                           "method", "method", VARCHAR(), true)},
+                  })
+                  .endTableScan()
+                  .filter("method = 'GET'")
+                  .planNode();
+
+  auto output =
+      getResults(plan, {makeClpSplit(getExampleFilePath("test_1.clps"))});
+  auto expected = makeRowVector(
+      {// requestId
+       makeFlatVector<StringView>(
+           {"req-100", "req-105", "req-107", "req-109", "req-102"}),
+       // userId
+       makeNullableFlatVector<StringView>(
+           {"user201", "user204", "user202", "user203", std::nullopt}),
+       // method
+       makeFlatVector<StringView>({
```

Comment:
```suggestion
       // requestId
       makeFlatVector<StringView>(
           {"req-100", "req-105", "req-107", "req-109", "req-102"}),

       // userId
       makeNullableFlatVector<StringView>(
           {"user201", "user204", "user202", "user203", std::nullopt}),

       // method
       makeFlatVector<StringView>({
```

Will the formatter allow this? If so, can you apply it in the other places?

Reply:
An extra newline is allowed, but after formatting, the `{` should remain on the same line as `// requestId`.

Reply:
Seems a bit weird to add a new line for other places

Reply:
Correct